src/allmydata/test/test_eliotutil.py

```diff
@@ -281,3 +282,75 @@ def f():
                 ),
             ),
         )
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_gets_args(self, logger):
+        """
+        Check that positional arguments are logged when using ``log_call_deferred``
+        """
+        @log_call_deferred(action_type=u"the-action")
+        def f(a):
+            return a ** 2
+        self.assertThat(
+            f(4), succeeded(Equals(16)))
+        msg = logger.messages[0]
+        assertContainsFields(self, msg, {"args": (4,)})
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_gets_kwargs(self, logger):
+        """
+        Check that keyword arguments are logged when using ``log_call_deferred``
+        """
+        @log_call_deferred(action_type=u"the-action")
+        def f(base, exp):
+            return base ** exp
+        self.assertThat(f(exp=2,base=10), succeeded(Equals(100)))
+        msg = logger.messages[0]
+        assertContainsFields(self, msg, {"kwargs": {"base": 10, "exp": 2}})
+
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_gets_kwargs_and_args(self, logger):
+        """
+        Check that both keyword and positional arguments are logged when using ``log_call_deferred``
+        """
+        @log_call_deferred(action_type=u"the-action")
+        def f(base, exp, message):
+            return base ** exp
+        self.assertThat(f(10, 2, message="an exponential function"), succeeded(Equals(100)))
+        msg = logger.messages[0]
+        assertContainsFields(self, msg, {"args": (10, 2)})
+        assertContainsFields(self, msg, {"kwargs": {"message": "an exponential function"}})
+
+
+    @capture_logging(
+        lambda self, logger:
+        assertHasAction(self, logger, u"the-action", succeeded=True),
+    )
+    def test_kwargs_dont_repeat_in_start_action(self, logger):
```

Comment:
I still don't understand what this test does.

Reply:
Well it tests that if we have ...

```python
@log_call_deferred(action_type=u"the-action")
def a_function(args, kwargs):
    pass

# function call with kwargs and args as arguments
a_function(kwargs=[], args=[])
```

Then, `kwargs` and `args` won't repeat or overlap in `eliotutil.log_call_deferred` where we do

https://github.com/Fenn-CS/tahoe-lafs/blob/aa00eb0bd4ef39449e5ecd67a8c2278cd2eb1752/src/allmydata/util/eliotutil.py#L338-#L339